cmd_migrate_db.go

```diff
@@ -0,0 +1,705 @@
+
+	return nil
+}
+
+type migrateDBCommand struct {
+	Source            *SourceDB `group:"source" namespace:"source" long:"" short:"" description:""`
+	Dest              *DestDB   `group:"dest" namespace:"dest" long:"" short:"" description:""`
+	Network           string    `long:"network" short:"n" description:"Network of the db files to migrate (used to navigate into the right directory)"`
+	PprofPort         int       `long:"pprof-port" description:"Enable pprof profiling on the specified port"`
+	ForceNewMigration bool      `long:"force-new-migration" description:"Force a new migration from the beginning of the source DB so the resume state will be discarded"`
+	ChunkSize         uint64    `long:"chunk-size" description:"Chunk size for the migration in bytes"`
+}
+
+func newMigrateDBCommand() *migrateDBCommand {
+	return &migrateDBCommand{
+		Source: &SourceDB{
+			Backend: lncfg.BoltBackend,
+			Bolt: &Bolt{
+				DBTimeout: kvdb.DefaultDBTimeout,
+				TowerDir:  defaultDataDir,
+				DataDir:   defaultDataDir,
+			},
+		},
+		Dest: &DestDB{
+			Backend:  lncfg.PostgresBackend,
+			Postgres: &postgres.Config{},
+			Sqlite: &Sqlite{
+				Config:   &sqlite.Config{},
+				TowerDir: defaultDataDir,
+				DataDir:  defaultDataDir,
+			},
+		},
+		Network: "mainnet",
+	}
+}
+
+func (x *migrateDBCommand) Register(parser *flags.Parser) error {
+	_, err := parser.AddCommand(
+		"migrate-db",
+		"Migrate the complete database state of lnd to a new backend",
+		`
+	Migrate the full database state of lnd from a source (for example the
+	set of bolt database files such as channel.db and wallet.db) database
+	to a SQL destination database.
+
+	IMPORTANT: Please read the data migration guide located	in the file
+	docs/data-migration.md of the main lnd repository before using this
+	command!
+
+	NOTE: The migration can take a long time depending on the amount of data
+	that needs to be written! The migration happens in chunks therefore it
+	can be resumed in case of an interruption. The migration also includes
+	a verification to assure that the migration is consistent.
+	As long as NEITHER the source nor destination database has been started/
+	run with lnd, the migration can be repeated/resumed in case of an error
+	since the data will just be overwritten again in the destination.
+
+	Once a database was successfully and completely migrated from the source
+	to the destination, the source will be marked with a 'tombstone' tag
+	while the destination will get an 'already migrated' tag.
+	A database with a tombstone cannot be started with lnd anymore to
+	prevent from an old state being used by accident.
+	To prevent overwriting a destination database by accident, the same
+	database/namespace pair cannot be used as the target of a data migration
+	twice, which is checked through the 'already migrated' tag.`,
+		x,
+	)
+	return err
+}
+
+func (x *migrateDBCommand) Execute(_ []string) error {
+	// TODO(ziggie): Add interceptor to cleanly shutdown the migration
```

Comment:
You can re-use lnd's signal package for this: https://pkg.go.dev/github.com/lightningnetwork/lnd@v0.18.5-beta.rc2/signal